src/main.rs

```diff
@@ -538,42 +541,47 @@ fn run(args: Cli) -> Result<i32> {
         printer.print_request_body(&mut request)?;
     }
 
+    let mut client = ClientWithMiddleware::new(client);
+
     if !args.offline {
         let mut response = {
             let history_print = args.history_print.unwrap_or(print);
-            let mut client = ClientWithMiddleware::new(&client);
-            if args.all {
-                client = client.with_printer(|prev_response, next_request| {
-                    if history_print.response_headers {
-                        printer.print_response_headers(prev_response)?;
-                    }
-                    if history_print.response_body {
-                        printer.print_response_body(
-                            prev_response,
-                            response_charset,
-                            response_mime,
-                        )?;
-                        printer.print_separator()?;
-                    }
-                    if history_print.response_meta {
-                        printer.print_response_meta(prev_response)?;
-                    }
-                    if history_print.request_headers {
-                        printer.print_request_headers(next_request, &*cookie_jar)?;
-                    }
-                    if history_print.request_body {
-                        printer.print_request_body(next_request)?;
-                    }
-                    Ok(())
-                });
-            }
             if args.follow {
                 client = client.with(RedirectFollower::new(args.max_redirects.unwrap_or(10)));
             }
             if let Some(Auth::Digest(username, password)) = &auth {
                 client = client.with(DigestAuthMiddleware::new(username, password));
             }
-            client.execute(request)?
+            #[cfg(target_family = "unix")]
+            if let Some(unix_socket) = args.unix_socket {
```

Comment:
There is a fork of [hyperlocal](https://github.com/Macil/hyperlocal-with-windows) that adds support for windows but it might be better to wait until it is merged upstream